vms/platformvm/validators_set_property_test.go

```diff
@@ -0,0 +1,853 @@
+	}
+	if err := vm.SetPreference(context.Background(), vm.manager.LastAccepted()); err != nil {
+		return nil, fmt.Errorf("failed verifying block, %s", err.Error())
+	}
+
+	return vm.state.GetCurrentValidator(stakerTx.SubnetID(), stakerTx.NodeID())
+}
+
+func terminateSubnetValidator(vm *VM, validator *state.Staker) error {
+	currentTime := validator.EndTime
+	vm.clock.Set(currentTime)
+	vm.state.SetTimestamp(currentTime)
+
+	blk, err := vm.Builder.BuildBlock(context.Background())
+	if err != nil {
+		return fmt.Errorf("failed building block, %s", err.Error())
+	}
+	if err := blk.Verify(context.Background()); err != nil {
+		return fmt.Errorf("failed verifying block, %s", err.Error())
+	}
+	if err := blk.Accept(context.Background()); err != nil {
+		return fmt.Errorf("failed accepting block, %s", err.Error())
+	}
+	if err := vm.SetPreference(context.Background(), vm.manager.LastAccepted()); err != nil {
+		return fmt.Errorf("failed verifying block, %s", err.Error())
+	}
+
+	// _, err = vm.state.GetCurrentValidator(constants.PrimaryNetworkID, nodeID)
+	// require.ErrorIs(err, database.ErrNotFound)
+	return nil
+}
+
+func terminatePrimaryValidator(vm *VM, validator *state.Staker) error {
+	currentTime := validator.EndTime
+	vm.clock.Set(currentTime)
+	vm.state.SetTimestamp(currentTime)
+
+	blk, err := vm.Builder.BuildBlock(context.Background())
+	if err != nil {
+		return fmt.Errorf("failed building block, %s", err.Error())
+	}
+	if err := blk.Verify(context.Background()); err != nil {
+		return fmt.Errorf("failed verifying block, %s", err.Error())
+	}
+
+	proposalBlk := blk.(snowman.OracleBlock)
+	options, err := proposalBlk.Options(context.Background())
+	if err != nil {
+		return fmt.Errorf("failed retrieving options, %s", err.Error())
+	}
+
+	commit := options[0].(*blockexecutor.Block)
+	_, ok := commit.Block.(*blocks.BanffCommitBlock)
+	if !ok {
+		return fmt.Errorf("failed retrieving commit option, %s", err.Error())
+	}
+	if err := blk.Accept(context.Background()); err != nil {
+		return fmt.Errorf("failed accepting block, %s", err.Error())
+	}
+
+	if err := commit.Verify(context.Background()); err != nil {
+		return fmt.Errorf("failed verifying commit block, %s", err.Error())
+	}
+	if err := commit.Accept(context.Background()); err != nil {
+		return fmt.Errorf("failed accepting commit block, %s", err.Error())
+	}
+
+	if err := vm.SetPreference(context.Background(), vm.manager.LastAccepted()); err != nil {
+		return fmt.Errorf("failed verifying block, %s", err.Error())
+	}
+
+	// _, err = vm.state.GetCurrentValidator(constants.PrimaryNetworkID, nodeID)
+	// require.ErrorIs(err, database.ErrNotFound)
+	return nil
+}
+
+// buildTimestampsList creates validators start and end time, given the event list.
+// output is returned as a list of state.Stakers, just because it's a convenient object to
+// collect all relevant information.
+// buildTimestampsList guarantees that start/end times of contiguous stakers do not overlap
```

Comment:
```suggestion
// buildTimestampsList creates validators start and end time, given the event list.
// output is returned as a list of state.Stakers, just because it's a convenient object to
// collect all relevant information.
```